domain/src/main/java/org/oppia/android/domain/topic/TopicListController.kt

```diff
@@ -161,101 +192,305 @@ class TopicListController @Inject constructor(
       .build()
   }
 
-  private fun createOngoingStoryListFromProgress(
+  private fun createUpcomingTopicSummaryFromJson(
+    topicId: String,
+    jsonObject: JSONObject
+  ): UpcomingTopic {
+    var totalChapterCount = 0
+    val storyData = jsonObject.getJSONArray("canonical_story_dicts")
+    for (i in 0 until storyData.length()) {
+      totalChapterCount += storyData
+        .getJSONObject(i)
+        .getJSONArray("node_titles")
+        .length()
+    }
+    val topicPlayAvailability = if (jsonObject.getBoolean("published")) {
+      TopicPlayAvailability.newBuilder().setAvailableToPlayNow(true).build()
+    } else {
+      TopicPlayAvailability.newBuilder().setAvailableToPlayInFuture(true).build()
+    }
+
+    return UpcomingTopic.newBuilder().setTopicId(topicId)
+      .setName(jsonObject.getString("topic_name"))
+      .setVersion(jsonObject.optInt("version"))
+      .setTopicPlayAvailability(topicPlayAvailability)
+      .setLessonThumbnail(createTopicThumbnail(jsonObject))
+      .build()
+  }
+
+  private fun computePromotedActivityList(
+    topicProgressList: List<TopicProgress>
+  ): PromotedActivityList {
+    val promotedActivityListBuilder = PromotedActivityList.newBuilder()
+    if (topicProgressList.isNotEmpty()) {
+      promotedActivityListBuilder.promotedStoryList = computePromotedStoryList(topicProgressList)
+      if (promotedActivityListBuilder.promotedStoryList.getTotalPromotedStoryCount() == 0) {
```

Comment:
```suggestion
      if (promotedActivityListBuilder.promotedStoryList.totalPromotedStoryCount == 0) {
```

Prefer to use Kotlin syntax for getters when available. Ditto elsewhere (wherever applicable).